### PR TITLE
feat: add validate contact form

### PR DESCRIPTION
**Pull request by Yinkusadel**
### This PR implement a client-side validation makes sure the content of the email field in the contact form is in lower case before submission

**Here is the link to the projects  [requirement page](https://github.com/bolah2009/curriculum-html-css/blob/main/portfolio/9_validate_form.md)**

### CODE REVIEW

File: js/validateinput.js [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+        return false;
+      }
+      fillFieldErrorMessage.innerText = '';
+    }
+  }
+
+  return true;
+};
+
+document.querySelector('.customer-contacts-form').addEventListener('submit', (event) => {
```
Comment (teckcipher) [inline js/validateinput.js]: I believe we can use the form `name` attribute here instead of the query selector.

**Review by moshoodabdullahi — CHANGES_REQUESTED** [review]
Hi @Yinkusadel Awesome work so far
I was going through the code and I figured that you didn't use the formspree, please try using the formspree mention by @teckcipher [Form Spree](https://help.formspree.io/hc/en-us/articles/360013470814-Submit-forms-with-JavaScript-AJAX-)

**Review by moshoodabdullahi — DISMISSED** [review]

File: index.html [hunk 2]
```diff
@@ -312,9 +312,14 @@ <h2>Skills</h2>
           <div class="customer-contacts d-flex justify-center">
             <form
               class="customer-contacts-form d-flex"
+              id="customer-contacts-form"
+              name="customerContactsForm"
               action="https://formspree.io/f/xzblbbzd"
               method="post"
+              required
```
Comment (teckcipher) [inline index.html]: Do we need this `required`?

File: js/validateinput.js [hunk 3]
```diff
@@ -0,0 +1,113 @@
...
+  const errorMessage = document.createElement('p');
+  errorMessage.classList.add('error-message');
+  errorMessage.textContent = message;
+
+  field.parentNode.insertBefore(errorMessage, field.nextSibling);
+};
+
+const form = document.getElementById('customer-contacts-form');
+
+form.addEventListener('submit', function (event) {
```
Comment (teckcipher) [inline js/validateinput.js]: Please, let's use a named function here. It makes debugging easier.

File: js/validateinput.js [hunk 4]
```diff
@@ -0,0 +1,113 @@
...
+
+const clearFormAndErrors = (isMobileView) => {
+  clearInputData(isMobileView);
+
+  clearErrors();
+};
+
+let isMobileView = window.innerWidth <= 767;
+
+window.addEventListener('resize', function () {
```
Comment (teckcipher) [inline js/validateinput.js]: same here

**Review by teckcipher — CHANGES_REQUESTED** [review]
Nice job with the logic used for validation, I left some comments, you can check them out and resend for another review.

**Review by teckcipher — APPROVED** [review]

**Review by moshoodabdullahi — APPROVED** [review]

**Pull request merged**